lib/screen_controller.dart

```diff
@@ -473,7 +473,15 @@ class ScreenController {
             );
             return;
           }
-          launchUrlString(uri, mode: LaunchMode.externalApplication);
+
+          uri = action.wallet == null
+              ? uri
+              : '${action.wallet?.toLowerCase().trim()}://wc?uri=$uri';
```

Comment:
do you have a complete list? if so force the action.wallet to be an enum and throw LanguageError if not in the list. Otherwise this is ok